File: java/javautil/collection/list/stack.java
import java.util.*;

public class stack {
    public static void main(String[] args){
        /* initialization */
        Stack<Integer> nums = new Stack<>();
        for (int i = 0; i < 10; i++){
            nums.add(i);
        }
        print("initialization", nums);

        /* empty */
        print("is empty", nums.empty());

        /* peek */
        print("peek", nums.peek());
        print("stack after peek", nums);

        /* pop */
        print("pop", nums.pop());
        print("stack after pop", nums);

        /* push */
        nums.push(10);
        print("stack after push", nums);

        /* search */
        print("the index of 5 in the stack", nums.search(5));
    }

    public static void print(String prompt, boolean is_empty){
        System.out.println(String.format("%-30s %-30s", prompt, is_empty));
    }

    public static void print(String prompt, Integer num){
        System.out.println(String.format("%-30s %-30s", prompt, num));
    }

    public static void print(String prompt, Stack<Integer> nums){
        System.out.println(String.format("%-30s %-30s", prompt, nums));
    }
}

File: data_structure/Stack/Stack.py
import typing


class Stack:
    def __init__(self, size: int):
        self.stk = []
        self.size = size
        self.top = -1

    def isEmpty(self):
        return self.top == -1
    
    def isFull(self):
        return self.top == self.size - 1
    
    def push(self, val: int):
        if self.isFull():
            raise Exception("Stack is full")
        else:
            self.stk.append(val)
            self.top += 1

    def pop(self):
        if self.isEmpty():
            raise Exception("Stack is empty")
        else:
            self.stk.pop()
            self.top -= 1

    def peek(self):
        if self.isEmpty():
            raise Exception("Stack is empty")
        else:
            return self.stk[self.top]
